iib/web/models.py

```diff
@@ -190,6 +204,79 @@ def add_state(self, state, state_reason):
         db.session.flush()
         self.request_state_id = request_state.id
 
+    def to_json(self, verbose=True):
+        states = [
+            {
+                'state': RequestStateMapping(state.state).name,
+                'state_reason': state.state_reason,
+                'updated': state.updated.isoformat(),
+            }
+            for state in self.states
+        ]
+        # Reverse the list since the latest states should be first
+        states = list(reversed(states))
+        latest_state = states[0]
+
+        rv = {
+            'id': self.id,
+            'bundles': [bundle.to_json() for bundle in self.bundles],
+            'binary_image': self.binary_image.to_json(),
+            'from_index': self.from_index.to_json(),
```

Comment:
In the API document, we said that all the images would be strings of the pull specification and not a JSON object. That is my preference to make the API simpler. This is especially true since we will not be recording the arches for images other than the built index image, so the empty `arches` list would be misleading.